### ISSUE TITLE
IDE0200 (lambda expression can be removed) results in compile error (ConditionalAttribute)

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#63464]
diagnostic is reported on:

https://github.com/dotnet/roslyn-analyzers/blob/793113a41d7f21b03470521bf48438f2abd9b12f/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysisData.cs#L83

![image](https://user-images.githubusercontent.com/31348972/185331916-69545955-6ead-419d-868b-aeca5b01bb09.png)

Looks like ConditionalAttribute needs a special handling.

Minimal repro:

```csharp
using System;
using System.Diagnostics;

public class C
{
    internal void M1()
    {
        M2(x => M3(x));
    }

    [Conditional("DEBUG")]
    internal void M3(string s) { }

    private static void M2(Action<string> a) { }
}
```
